nni/compression/experiment/config/vessel.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+import base64
+import io
+from dataclasses import dataclass, asdict
+from pathlib import Path
+from typing import Any, Callable, Tuple
+from typing_extensions import Literal
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from nni.algorithms.compression.v2.pytorch.utils.constructor_helper import OptimizerConstructHelper
+from nni.common import dump, load
+from nni.experiment.config.base import ConfigBase
+
+
+@dataclass(init=False)
+class CompressionVessel(ConfigBase):
+    model: str | bytes
+    finetuner: str | bytes
+    evaluator: str | bytes
+    dummy_input: str | bytes
+    trainer: str | bytes | Literal['null'] | None
+    optimizer_helper: str | bytes | Literal['null'] | None
+    criterion: str | bytes | Literal['null'] | None
+    device: str
+
+    def __init__(self,
+                 model: Module,
```

Comment:
So `model` should be `Module | str`.

Reply:
will add an @overload `__init__` for `str` input, in fact, these `str` interface does not want to be exposed to users

Reply:
`@overload` is meant for type hint.
https://docs.python.org/3/library/typing.html#typing.overload

Reply:
`@overload` is meant for type hint.
https://docs.python.org/3/library/typing.html#typing.overload

Reply:
yes, type hint is all that needed.